picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl

```diff
@@ -0,0 +1,166 @@
+      them optical duplicates. The default is appropriate for unpatterned
+      versions of the Illumina platform. For the patterned flowcell models, 2500
+      is moreappropriate. For other platforms and models, users should
+      experiment to find what works best.  Default value: 100. This option can
+      be set to 'null' to clear the default value.
+outputs:
+  - id: bam
+    type: File
+    outputBinding:
+      glob: '$(inputs.input.basename.replace(/.bam/, ''_md.bam''))'
+    secondaryFiles:
+      - ^.bai
+      - ^.metrics
+label: picard_mark_duplicates_2.8.1
+arguments:
+  - position: 0
+    prefix: ''
+    valueFrom: "${\n  if(inputs.memory_per_job && inputs.memory_overhead) {\n   \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if (inputs.memory_per_job && !inputs.memory_overhead){\n    \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if(!inputs.memory_per_job && inputs.memory_overhead){\n    \n    return \"-Xmx15G\"\n  }\n  else {\n    \n  \treturn \"-Xmx15G\"\n  }\n}"
+  - position: 0
+    prefix: '-jar'
+    valueFrom: /usr/local/bin/picard.jar
+  - position: 0
+    prefix: ''
+    valueFrom: MarkDuplicates
+requirements:
+  - class: ResourceRequirement
+    ramMin: "${\r  if(inputs.memory_per_job && inputs.memory_overhead) {\r   \r    return inputs.memory_per_job + inputs.memory_overhead\r  }\r  else if (inputs.memory_per_job && !inputs.memory_overhead){\r    \r   \treturn inputs.memory_per_job + 2000\r  }\r  else if(!inputs.memory_per_job && inputs.memory_overhead){\r    \r    return 15000 + inputs.memory_overhead\r  }\r  else {\r    \r  \treturn 17000 \r  }\r}"
+    coresMin: "${\r  if (inputs.number_of_threads) {\r    \r   \treturn inputs.number_of_threads \r  }\r  else {\r    \r    return 2\r  }\r}"
+  - class: DockerRequirement
+    dockerPull: 'mskcc/picard:2.8.1'
+  - class: InlineJavascriptRequirement
+'dct:contributor':
+  - class: 'foaf:Organization'
+    'foaf:member':
+      - class: 'foaf:Person'
+        'foaf:mbox': 'mailto:shahr2@mskcc.org'
+        'foaf:name': Ronak Shah
+    'foaf:name': Memorial Sloan Kettering Cancer Center
+'dct:creator':
+  - class: 'foaf:Organization'
+    'foaf:member':
+      - class: 'foaf:Person'
+        'foaf:mbox': 'mailto:shahr2@mskcc.org'
+        'foaf:name': Ronak Shah
+    'foaf:name': Memorial Sloan Kettering Cancer Center
+'doap:release':
+  - class: 'doap:Version'
+    'doap:name': picard
+    'doap:revision': 2.8.1
+  - class: 'doap:Version'
+    'doap:name': cwl-wrapper
+    'doap:revision': 1.0.0
```

Comment:
would be nice to get rabix to put this at the top of the CWL but it's a minor thing I think

Reply:
![bitmoji](https://render.bitstrips.com/v2/cpanel/12172113-833e-4569-bcfd-c0297ac8c4aa-091819f5-3b43-4653-8093-982c9e72a960-v1.png?transparent=1&palette=1&width=246)